rfcs/006 - Replicate changes between staging and prod RFC.md

```diff
@@ -0,0 +1,87 @@
+
+`INSERT` statements that are referencing an auto-generated ID should generate the SQL with a sub-query that `SELECT`s on the @cloudId column's value, if available. It should be immediately clear to the user that they need to substitute that string with the production equivalent ID, and in some situations where the cloud ID is a user-provided string, it may not need any changes.
+
+## Alternatives Considered
+
+There are several types of solutions to replicate staging infrastructure changes in production.
+
+### 1. Dump a database and import the dump into another database
+
+An `export` route in the engine implemented using `pgdump` with logic to attempt to make it generic like excluding `aws_account` from the dump and so on. The dump is then intended to be passed into an `import` route for another database. `import` was deprecated for the time being due to the sheer amount of special-casing, which compounded per module, that had to be built within the functionality of export and import.
+
+### 2. Sync a database to a different cloud account than the one it is connected to
+
+An IaSQL function that takes the alias of another dataabse owned by the same user as input. `iasql_import(‘staging’)` will return the same data structure as `iasql_commit` or `iasql_preview`. This function can reuse parts of the mappers and existing IaSQL functions to run diffs across two different databases or accounts. There could have an optional `dry_run` parameter or another function to preview the changes: `iasql_preview_import(‘staging’)` or `iasql_import(‘staging’, true)`.
+
+The steps to replicate recent database/account A changes in database/account B when calling `iasql_import('A')` from within database B:
+1) Regular sync database A to cloud account A
+2) Regular sync database B to cloud account B
+3) New import sync functionality where database B is pointed to cloud account A directly
+
+The new import sync functionality described in #3 would expand the mapper definitions to include a new `import` function and laxer equality function for each module.
+
+### 3. Interactive `git rebase` style sync
+
+Similar to option 2, but when conflict arises prompt the user to handle the merge conflicts between the two cloud accounts or databases interactively. The PostgreSQL REPL doesn't allow interactive input so that would have to happen using a two-phase process via an interim table or the dashboard. This option requires a lot of resources to implement out of the gate. The current proposal and if this workflow proves to be primordial to users we can eventually prioritize it.
```

Comment:
```suggestion
Similar to option 2, but when conflict arises prompt the user to handle the merge conflicts between the two cloud accounts or databases interactively. The PostgreSQL REPL doesn't allow interactive input so that would have to happen using a two-phase process via an interim table or the dashboard. This option requires a lot of resources to implement out of the gate. The current proposal and this workflow are compatible, so if this proves to be important to users we can eventually prioritize it.
```